lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -371,14 +374,46 @@ public void indexObjects(
                 pw.println(UObject.transformObjectToString(doc));
             }
             pw.close();
-            makeRequestBulk("POST", indexName, tempFile);
+            Response response = makeRequestBulk("POST", indexName, tempFile);
+
+            verify(response);
+
             updateLastVersionsInData(indexName, pguid, lastVersion);
         } finally {
             tempFile.delete();
         }
         refreshIndex(indexName);
     }
-    
+
+
+    private void verify(Response response) throws IOException {
+        // convert JSON string to Map
+        String responseJsonStr = EntityUtils.toString(response.getEntity());
+
+        ObjectMapper mapper = new ObjectMapper();
+        final Map<String, Object> responseMap = mapper.readValue(responseJsonStr,
+                new TypeReference<Map<String, Object>>() {});
+
+        // check for and collect response data from index operations that failed
+        List errorItems = new ArrayList();
+        if(responseMap.get("errors").equals(Boolean.TRUE)) {
+            for(Map item: (List<Map>)responseMap.get("items")) {
+                for(Object itemData: item.values()) {
+                    if (!((Map) itemData).get("status")
+                            .toString().startsWith("20")) {  // status code 200, 201 etc
+                        errorItems.add(itemData);
+                    }
+                }
+            }
+        }
+
+        // report response data from failed index operations
+        if(errorItems.size() > 0) {
+            throw new IOException("Indexing errors: " + errorItems.toString());
```

Comment:
Doesn't seem like there's any tests for this

Reply:
SearchAPIIntegrationTest.strictMappingFail() covers this case.

Reply:
Ok, so the test really tests 2 separate things - can you note that in comments in the test?

Reply:
Not sure I follow. I am only testing one assertion, which has now changed to verifying the log statement and I have added a comment for this.